### ISSUE TITLE
Eclipse 2021-06 - Unable to make private... error

### ISSUE THREAD

**Issue by tullio0106** [eclipse-m2e/m2e-core#247]
I migrated to Eclipse 2021-06.
Opening a project I get an error :
Unable to make field private final java.util.Comparator java.util.TreeMap.comparator accessible: module java.base does not "opens java.util" to unnamed module @73901b53
on a pom file.
What's wrong ?
The project still opens in 2021-03 without errors.
Tks

**Comment by mickaelistria** [comment]
Can you please provide details of how to reproduce the issue? Where is the error shown?

**Comment by tullio0106** [comment]
The error is shown in the Problems tab.
The only relevant message in the errore log window is
eclipse.buildId=4.20.0.I20210611-1600
java.version=16.0.1
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=it_IT
Framework arguments:  -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product

org.eclipse.m2e.logback.appender
Warning
Tue Jun 22 11:46:47 CEST 2021
Error injecting: org.apache.maven.plugin.war.WarMojo

com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, java.lang.ExceptionInInitializerError
  at org.apache.maven.plugin.war.WarMojo.<init>(Unknown Source)
  while locating org.apache.maven.plugin.war.WarMojo

1 error
	at com.google.inject.internal.InternalProvisionException.toProvisionException(InternalProvisionException.java:226)
	at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1053)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1086)
	at org.eclipse.sisu.space.AbstractDeferredClass.get(AbstractDeferredClass.java:48)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:85)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:57)
	at com.google.inject.internal.ProviderInternalFactory$1.call(ProviderInternalFactory.java:66)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:127)
	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:47)
	at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
	at org.eclipse.sisu.inject.Guice4$1.get(Guice4.java:162)
	at org.eclipse.sisu.inject.LazyBeanEntry.getValue(LazyBeanEntry.java:81)
	at org.eclipse.sisu.plexus.LazyPlexusBean.getValue(LazyPlexusBean.java:51)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:263)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:255)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo(DefaultMavenPluginManager.java:520)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl.getConfiguredMojo(MavenImpl.java:354)
	at org.sonatype.m2e.mavenarchiver.internal.AbstractMavenArchiverConfigurator.reflectManifestGeneration(AbstractMavenArchiverConfigurator.java:469)
	at org.sonatype.m2e.mavenarchiver.internal.AbstractMavenArchiverConfigurator.generateManifest(AbstractMavenArchiverConfigurator.java:422)
	at org.sonatype.m2e.mavenarchiver.internal.AbstractMavenArchiverConfigurator.mavenProjectChanged(AbstractMavenArchiverConfigurator.java:227)
	at org.sonatype.m2e.mavenarchiver.internal.AbstractMavenArchiverConfigurator$1.build(AbstractMavenArchiverConfigurator.java:167)
	at org.eclipse.m2e.core.internal.builder.MavenBuilderImpl.build(MavenBuilderImpl.java:135)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$1.method(MavenBuilder.java:169)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$1.method(MavenBuilder.java:1)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.lambda$1(MavenBuilder.java:114)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:179)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:114)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.lambda$0(MavenBuilder.java:105)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:179)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:153)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:101)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.execute(MavenBuilder.java:88)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder.build(MavenBuilder.java:197)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:846)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:229)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:277)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:330)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:333)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:385)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:406)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.ExceptionInInitializerError
	at com.thoughtworks.xstream.XStream.setupConverters(XStream.java:678)
	at com.thoughtworks.xstream.XStream.<init>(XStream.java:456)
	at com.thoughtworks.xstream.XStream.<init>(XStream.java:381)
	at com.thoughtworks.xstream.XStream.<init>(XStream.java:356)
	at org.apache.maven.plugin.war.util.WebappStructureSerializer.<clinit>(WebappStructureSerializer.java:47)
	at org.apache.maven.plugin.war.AbstractWarMojo.<init>(AbstractWarMojo.java:345)
	at org.apache.maven.plugin.war.WarMojo.<init>(WarMojo.java:53)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at com.google.inject.internal.DefaultConstructionProxyFactory$ReflectiveProxy.newInstance(DefaultConstructionProxyFactory.java:126)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
	at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:127)
	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
	at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
	... 46 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.util.Comparator java.util.TreeMap.comparator accessible: module java.base does not "opens java.util" to unnamed module @12cd43e7
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:177)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:171)
	at com.thoughtworks.xstream.converters.collections.TreeMapConverter.<clinit>(TreeMapConverter.java:59)
	... 68 more

**Comment by mickaelistria** [comment]
Can you please share a minimal project to reproduce this error?

**Comment by tullio0106** [comment]
Create a small project is quite difficult becuase it's part of a very large project.

Is there any different way to proceed ?

Tks



 


	
		
			 
			
			
				
					
						Dott. Tullio Bettinazzi
					
				
			

			
				
					
						Responsabile R&amp;D
					
				
			
			
			 
		
		
			 
			
			 
		
		
			 
			
			
				
					
						Axioma Srl
					
				
			

			
				
					
						Via Gerardo e Antonio Philips, 12
					
					
						20900 Monza (MB)
					
					
						***@***.***
					
					
						+39 03922511
					
				
			

			
				
					
						+39 335 1048626
					
				
			

			
				
					
						www.axioma.it
					
				
			
			
			 
		
		
			 
			
			
				
					
						
						
						
					
				
			
			
			 
		
		
			 
		
		
			 
			The content of this e-mail and any attachments, is strictly confidential, not to be produced in court and intended for the person (s) to whom it is addressed. If you have received this email in error, please let us know immediately and delete it. It is forbidden to copy and disclose the contents of this email. Any unauthorized use of the information contained herein by third parties or in any case not indicated in this email may be prosecuted according to the law. We inform you that for the exercise of the rights provided for by the articles 15 and ss. of the 2016/679 EU Regulation (GDPR), you can contact the following address: ***@***.***
			 
		
	




 


----- Messaggio Originale -----

Da: "Mickael Istria" ***@***.***&gt;

A: "eclipse-m2e/m2e-core" ***@***.***&gt;

Cc: "tullio0106" ***@***.***&gt;, "Author" ***@***.***&gt;

Inviato: 30/06/2021 17:10

Oggetto: Re: [eclipse-m2e/m2e-core] Eclipse 2021-06 - Unable to make private... error (#247)

 


 



Can you please share a minimal project to reproduce this error?



—

You are receiving this because you authored the thread.

Reply to this email directly, view it on GitHub, or unsubscribe.






--

This message has been checked by Libraesva ESG and is found to be clean.

Mark it as spam

Blacklist sender

**Comment by mickaelistria** [comment]
> Create a small project is quite difficult becuase it's part of a very large project.

Not really, I don't imagine anyone who is not facing this particular issue being very enthusiast in making extra effort to reproduce it and reduce the faulty case. Usually, in OSS projects, the people who face the issue are expected to provide a minimal reproducer if they want more chance for their issues to be considered.
Basically, the less you give, the less you have.

**Comment by tullio0106** [comment]
I understand but if You have a small test application already done I can test it in my environment and see if the problem reproduces.

Tks



 


	
		
			 
			
			
				
					
						Dott. Tullio Bettinazzi
					
				
			

			
				
					
						Responsabile R&amp;D
					
				
			
			
			 
		
		
			 
			
			 
		
		
			 
			
			
				
					
						Axioma Srl
					
				
			

			
				
					
						Via Gerardo e Antonio Philips, 12
					
					
						20900 Monza (MB)
					
					
						***@***.***
					
					
						+39 03922511
					
				
			

			
				
					
						+39 335 1048626
					
				
			

			
				
					
						www.axioma.it
					
				
			
			
			 
		
		
			 
			
			
				
					
						
						
						
					
				
			
			
			 
		
		
			 
		
		
			 
			The content of this e-mail and any attachments, is strictly confidential, not to be produced in court and intended for the person (s) to whom it is addressed. If you have received this email in error, please let us know immediately and delete it. It is forbidden to copy and disclose the contents of this email. Any unauthorized use of the information contained herein by third parties or in any case not indicated in this email may be prosecuted according to the law. We inform you that for the exercise of the rights provided for by the articles 15 and ss. of the 2016/679 EU Regulation (GDPR), you can contact the following address: ***@***.***
			 
		
	




 


----- Messaggio Originale -----

Da: "Mickael Istria" ***@***.***&gt;

A: "eclipse-m2e/m2e-core" ***@***.***&gt;

Cc: "tullio0106" ***@***.***&gt;, "Author" ***@***.***&gt;

Inviato: 30/06/2021 17:47

Oggetto: Re: [eclipse-m2e/m2e-core] Eclipse 2021-06 - Unable to make private... error (#247)

 


 




Create a small project is quite difficult becuase it's part of a very large project.




Not really, I don't imagine anyone who is not facing this particular issue being very enthusiast in making extra effort to reproduce it and reduce the faulty case. Usually, in OSS projects, the people who face the issue are expected to provide a minimal reproducer if they want more chance for their issues to be considered.

Basically, the less you give, the less you have.



—

You are receiving this because you authored the thread.

Reply to this email directly, view it on GitHub, or unsubscribe.






--

This message has been checked by Libraesva ESG and is found to be clean.

Mark it as spam

Blacklist sender

**Comment by cal101** [comment]
Just a wild guess:

You seem to be using java.version=16.0.1.
Is it possible that the version of  the war plugin used (org.apache.maven.plugin.war.WarMojo) is not compatible with this?
I am not sure if Java 16 is the runtime for eclipse and if eclipse is loading the WarMojo in its own JVM.
If yes I would either try updating the plugin or trying to run eclipse with an older JRE if that is possible.

If executing the command line mvn with java 16 runtime leads to the same problem that would indicate a problem of the War-Plugin with java 16.

**Comment by tullio0106** [comment]
Eclipse runs in java 11 but the plugin produces a war.

If I run my application (I use Java 8 as JVM) the application runs perfectly in a non touch environment, in a touch environment exhibits this strange behaviour (the same war) :

	Chrome 91 does'nt work at all as well as Edge
	Firefox 89 works only using localhost as server in the url, the same server, if I use IP address in the url, it doesn't work
	MSIE works perfectly
	An old Chrome version on an ol Ipad it works

So it seems to be a "browser related" problem.

Tks



 


	
		
			 
			
			
				
					
						Dott. Tullio Bettinazzi
					
				
			

			
				
					
						Responsabile R&amp;D
					
				
			
			
			 
		
		
			 
			
			 
		
		
			 
			
			
				
					
						Axioma Srl
					
				
			

			
				
					
						Via Gerardo e Antonio Philips, 12
					
					
						20900 Monza (MB)
					
					
						***@***.***
					
					
						+39 03922511
					
				
			

			
				
					
						+39 335 1048626
					
				
			

			
				
					
						www.axioma.it
					
				
			
			
			 
		
		
			 
			
			
				
					
						
						
						
					
				
			
			
			 
		
		
			 
		
		
			 
			The content of this e-mail and any attachments, is strictly confidential, not to be produced in court and intended for the person (s) to whom it is addressed. If you have received this email in error, please let us know immediately and delete it. It is forbidden to copy and disclose the contents of this email. Any unauthorized use of the information contained herein by third parties or in any case not indicated in this email may be prosecuted according to the law. We inform you that for the exercise of the rights provided for by the articles 15 and ss. of the 2016/679 EU Regulation (GDPR), you can contact the following address: ***@***.***
			 
		
	




 


----- Messaggio Originale -----

Da: "cal" ***@***.***&gt;

A: "eclipse-m2e/m2e-core" ***@***.***&gt;

Cc: "tullio0106" ***@***.***&gt;, "Author" ***@***.***&gt;

Inviato: 30/06/2021 18:55

Oggetto: Re: [eclipse-m2e/m2e-core] Eclipse 2021-06 - Unable to make private... error (#247)

 


 



Just a wild guess:



You seem to be using java.version=16.0.1.

Is it possible that the version of the war plugin used (org.apache.maven.plugin.war.WarMojo) is not compatible with this?

I am not sure if Java 16 is the runtime for eclipse and if eclipse is loading the WarMojo in its own JVM.

If yes I would either try updating the plugin or trying to run eclipse with an older JRE if that is possible.



If executing the command line mvn with java 16 runtime leads to the same problem that would indicate a problem of the War-Plugin with java 16.



—

You are receiving this because you authored the thread.

Reply to this email directly, view it on GitHub, or unsubscribe.






--

This message has been checked by Libraesva ESG and is found to be clean.

Mark it as spam

Blacklist sender

**Comment by cpierceworld** [comment]
I have the same problem...

It happens running Eclipse with Java 11+ (required for Eclipse 2021-06) but project uses Java 8.

i.e. similar to #248, and before that [https://bugs.eclipse.org/bugs/show_bug.cgi?id=432992](https://bugs.eclipse.org/bugs/show_bug.cgi?id=432992)

**Comment by djaclasa** [comment]
I have the same escenario and works for me :

migrate the version for war plugin: from 3.0.0 to 3.3.1

<plugin>
	<artifactId>maven-war-plugin</artifactId>
	<version>3.3.1</version>
	<configuration>
	<warSourceDirectory>WebContent</warSourceDirectory>
	</configuration>
</plugin>

**Comment by tullio0106** [comment]
Dott. Tullio Bettinazzi 
          
         
        
        
         
          
          Responsabile R&amp;D 
          
         
         
        
      
      
        
       
        
      
      
        
       
        
         
          
          Axioma Srl 
          
         
        
        
         
          
          Via Gerardo e Antonio Philips, 12 
          
          
          20900 Monza (MB) 
          
          
          ***@***.*** 
          
          
          +39 03922511 
          
         
        
        
         
          
          +39 335 1048626 
          
         
        
        
         
          
          www.axioma.it 
          
         
         
        
      
      
        
       
        
         
          
           
           
           
          
         
         
        
      
      
        
      
      
        
      The content of this e-mail and any attachments, is strictly confidential, not to be produced in court and intended for the person (s) to whom it is addressed. If you have received this email in error, please let us know immediately and delete it. It is forbidden to copy and disclose the contents of this email. Any unauthorized use of the information contained herein by third parties or in any case not indicated in this email may be prosecuted according to the law. We inform you that for the exercise of the rights provided for by the articles 15 and ss. of the 2016/679 EU Regulation (GDPR), you can contact the following address: ***@***.*** 
        
      
     
    
   

  
  
  
   ----- Messaggio Originale -----
   
Da: "djaclasa" ***@***.***&gt;
   
A: "eclipse-m2e/m2e-core" ***@***.***&gt;
   
Cc: "tullio0106" ***@***.***&gt;, "Author" ***@***.***&gt;
   
Inviato: 24/02/2022 15:40
   
Oggetto: Re: [eclipse-m2e/m2e-core] Eclipse 2021-06 - Unable to make private... error (#247)
   

   

   


   

I have the same escenario and works for me :
   

migrate the version for war plugin: from 3.0.0 to 3.3.1 maven-war-plugin 3.3.1 WebContent 
   

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
   
   


 -- 
 This message has been checked by Libraesva ESG and is found to be clean.
 Mark it as spam
 Blacklist sender
